cloudify_cli/execution_events_fetcher.py

```diff
@@ -112,18 +145,21 @@ def wait_for_execution(client,
                                        execution.deployment_id))
             else:
                 # update the remaining timeout
-                timeout = deadline-time.time()
+                timeout = deadline - time.time()
 
-        if execution.status != Execution.PENDING:
-            events_fetcher.fetch_and_process_events(
-                events_handler=events_handler, timeout=timeout)
+        if not execution_ended:
+            execution = client.executions.get(execution.id)
+            if execution.status in Execution.END_STATES:
```

Comment:
I think execution_ended = execution.status in Execution.END_STATES could simplify things, what do you think?